### ISSUE TITLE
What happened when softmax is applied to the output of classifiers?

### ISSUE THREAD

**Issue by hermosayhl** [HuiZeng/Image-Adaptive-3DLUT#15]
When the 3D-LUT algorithm is applied to one image, I found that the output of classifiers is not normalized.
**For example**
`weights  :  tensor([ 3.3231, -3.0477, -0.2689]`
**When softmax is applied**
![image](https://user-images.githubusercontent.com/49510831/95707305-bd3fe200-0c8b-11eb-9c3e-8941298d6c5b.png)

![image](https://user-images.githubusercontent.com/49510831/95701228-97aadc80-0c7b-11eb-975a-758119cbb447.png)
The training process collapsed.

![image](https://user-images.githubusercontent.com/49510831/95701330-e2c4ef80-0c7b-11eb-9244-4feb5f080a44.png)

**Where does it go wrong?**

**Comment by HuiZeng** [comment]
Hi, I have tried both softmax and sigmoid normalization, both of which lead to worse performance. 
There are two issues here:
First, normalization should be put on the fused 3D LUT rather than predicted weights. Normalizing weights may hurt the flexibility of the resulted LUT.
Second, both softmax and sigmoid are easy to saturate. Once saturated, they are hard to update because of very small gradients. You can plot the predicted weights during training process to check the details.
